src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -865,6 +865,12 @@ public static ITypeSymbol GetSymbolType(this ISymbol symbol)
                 return aliasSymbol.Target as ITypeSymbol;
             }
 
+            var methodSymbol = symbol as IMethodSymbol;
```

Comment:
Sorry, don't quite follow. Do the implemetors of IMethodSymbol also implement ITypeSymbol to make that happen?